### PR TITLE
add same site property to cookie

### PR DESCRIPTION
**Pull request by Dan-Y-Ko**
# Description

I'm having trouble getting a screenshot for this but basically there is an issue with the cookies and cross-site domains so we need to set the same-site as none. Adding that fixed this issue

This is probably only an issue with client components but since we are gunna have a separate react project in the future (which would be client) that'll communicate with this backend, this would fix for that usecase as well

## Issue link

Fixes # (issue)

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Feature updates / changes
- [ ] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

I ran into this issue when trying to test from localhost on frontend and the deployed backend dev server.
I also ran into this issue using localhost on the backend and the preview deployed url on vercel (it's using the localhost backend, just different domain).

After implementing this, it worked as expected

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

File: src/auth/auth.service.ts [hunk 1]
```diff
@@ -134,11 +134,13 @@ export class AuthService {
             maxAge: AT_MAX_AGE * 1000,
             httpOnly: true,
             secure: true,
+            sameSite: "none",
```
Comment (cherylli) [inline src/auth/auth.service.ts:137]:
```suggestion
            sameSite: "none", // TODO: remove this in future when we use actual domains for the dashboard, they will likely be same-site
```

Comment (cherylli) [reply src/auth/auth.service.ts:137]: just want to add a note here so we don't forget
I imagine the dashboard could be something like dashboard.chingu.io, and backend would be api.chingu.io

**Review by cherylli — DISMISSED** [review]

**Comment by Dan-Y-Ko** [comment]
I do want to point out that if you remove this later, we won't be able to use the deployed be dev server and we will have to use the local dev server. I don't really want our team to be forced to do that tbh

**Comment by cherylli** [comment]
> I do want to point out that if you remove this later, we won't be able to use the deployed be dev server and we will have to use the local dev server. I don't really want our team to be forced to do that tbh

you are right, maybe we need to have a condition statement for dev/prod environment.

File: src/auth/auth.service.ts [hunk 2]
```diff
@@ -134,11 +134,13 @@ export class AuthService {
             maxAge: AT_MAX_AGE * 1000,
             httpOnly: true,
             secure: true,
+            sameSite: "none", // TODO: remove this in future when we use actual domains for the dashboard, they will likely be same-site
```
Comment (cherylli) [inline src/auth/auth.service.ts:137]:
```suggestion
            sameSite: "none",
```

Comment (cherylli) [reply src/auth/auth.service.ts:137]: remove this for now and put on click up instead, I need to do more research on how it is usually handled. 

I'm thinking set cookies based on the environment, i.e. NODE_ENV

**Review by cherylli — APPROVED** [review]

**Review by timDeHof — APPROVED** [review]
All tests passed. But I think the `returnTokensOnLoginSuccess` should be wrapped in a try-catch for error handling. Also it should have a explicit return type for clarity and good practices.

**Pull request merged**